### PR TITLE
VBLOCKS-1659 Dialer Screen Component Tests.

### PR DESCRIPTION
**Pull request by manjeshbhargav**
## Submission Checklist

 - [ ] The `CHANGELOG.md` reflects any **feature**, **bug fixes**, or **known issues** made in the source code
 - [x] New unit tests and integration tests have beed added
 - [x] Code coverage is more or equal to the previous coverage
 - [x] A visual inspection of the `Files changed` tab was made prior to submitting the pull request ensuring the style guide was followed
 - [x] [CI pipeline](https://app.circleci.com/pipelines/github/twilio/twilio-voice-react-native-app) is green
 - [ ] Included screenshot if necessary

## Description

Component tests for the Dialer screen as specified [here](https://docs.google.com/document/d/1RrlgVcNIgGT2JRu_ClI1EoigfaXaq1zXy1AIsxR-ZHQ/edit#heading=h.pxgth5j86y4q).

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: app/src/screens/Dialer/__tests__/Dialer.test.tsx [hunk 1]
```diff
@@ -25,50 +26,195 @@ describe('<Dialer />', () => {
...
-    fireEvent.press(screen.getByText('4'));
-    fireEvent.press(screen.getByText('5'));
-    fireEvent.press(screen.getByText('6'));
-    fireEvent.press(screen.getByText('3'));
-    fireEvent.press(screen.getByText('8'));
-    expect(screen.getByText('+4157245638')).toBeOnTheScreen();
-  });
+    describe('UI layout', () => {
+      it('should display the enabled dialpad buttons', () => {
+        ['1', '2', '3', '4', '5', '6', '7', '8', '9', '*', '0', '#'].forEach(
```
Comment (kpchoy) [inline app/src/screens/Dialer/__tests__/Dialer.test.tsx:36]: 👍

File: app/src/screens/Dialer/__tests__/Dialer.test.tsx [hunk 2]
```diff
@@ -25,50 +26,195 @@ describe('<Dialer />', () => {
...
 
-  it('should allow user to use the backspace button', () => {
-    render(<Dialer />, { wrapper });
-    fireEvent.press(screen.getByText('4'));
-    fireEvent.press(screen.getByText('1'));
-    fireEvent.press(screen.getByText('5'));
-    expect(screen.getByText('+415')).toBeOnTheScreen();
-    fireEvent.press(screen.getByTestId('back_space_button'));
-    expect(screen.getByText('+41')).toBeOnTheScreen();
+      describe('press the backspace button', () => {
```
Comment (kpchoy) [inline app/src/screens/Dialer/__tests__/Dialer.test.tsx:118]: do you still need this "backspace button" test with the "backspace button" tests on line 82-101?

**Review by kpchoy — COMMENTED** [review]
Looking good so far 👍

File: app/src/screens/Dialer/__tests__/Dialer.test.tsx (hunk 2 above)
Comment (manjeshbhargav) [reply app/src/screens/Dialer/__tests__/Dialer.test.tsx:149]: @kpchoy Those are two different scenarios:
1. Press backspace when there is only one digit on display
2. Press backspace when there are more than one digit on display

File: app/package.json [hunk 3]
```diff
@@ -49,6 +49,7 @@
     "metro-react-native-babel-preset": "0.72.3",
     "react-test-renderer": "18.1.0",
     "redux-mock-store": "^1.5.4",
+    "redux-mock-store-await-actions": "^2.1.0",
```
Comment (mhuynh5757) [inline app/package.json:52]: This package is deprecated, let's consider writing the functionality ourselves using something similar to the store tests and polling.

File: app/src/screens/Dialer/__tests__/Dialer.test.tsx [hunk 4]
```diff
@@ -1,74 +1,291 @@
 import '@testing-library/jest-native/extend-expect';
 import { render, screen, fireEvent } from '@testing-library/react-native';
 import React from 'react';
+import { Text } from 'react-native';
 import { Provider } from 'react-redux';
-import configureStore from 'redux-mock-store';
+import configureStore, { MockStore } from 'redux-mock-store';
```
Comment (kpchoy) [inline app/src/screens/Dialer/__tests__/Dialer.test.tsx:6]: I thought we were going to not use `redux-mock-store` ?

What if you use `createStore` from https://github.com/twilio/twilio-voice-react-native-app/blob/main/app/src/store/app.ts#L11 ?

Comment (mhuynh5757) [reply app/src/screens/Dialer/__tests__/Dialer.test.tsx:6]: I spoke with @manjeshbhargav about this and it seems like there were unforeseen issues. I'll let @manjeshbhargav speak to it. It is unfortunate that we could not do it with this PR. 😞

File: app/src/screens/Dialer/__tests__/Dialer.test.tsx [hunk 5]
```diff
@@ -1,74 +1,291 @@
...
+    describe('press more dialpad buttons', () => {
+      beforeEach(() => {
+        ['4', '1', '5', '6', '5', '0', '2', '8', '9', '0'].forEach((digit) => {
+          fireEvent.press(screen.getByText(digit));
+        });
+      });
+
+      it('should append the pressed dialpad digits to the display', () => {
+        expect(screen.getByTestId('formatted_number').props.children).toBe(
+          '+4156502890',
```
Comment (mhuynh5757) [inline app/src/screens/Dialer/__tests__/Dialer.test.tsx:173]: Just curious what number is this lol

**Review by mhuynh5757 — APPROVED** [review]
Looks good to me, let's note down somewhere that there's potential improvement for the `waitForActions` function using `Promise.all` and some custom thunk handler, and using the actual store.

File: app/src/screens/Dialer/__tests__/Dialer.test.tsx (hunk 4 above)
Comment (manjeshbhargav) [reply app/src/screens/Dialer/__tests__/Dialer.test.tsx:6]: @kpchoy I tried integrating the real store, and there were a lot of issues related to the react dom that I don't have the expertise yet to debug. I would love to work on this in a separate initiative.

File: app/src/screens/Dialer/__tests__/Dialer.test.tsx (hunk 5 above)
Comment (manjeshbhargav) [reply app/src/screens/Dialer/__tests__/Dialer.test.tsx:173]: @mhuynh5757 Some random number. If its a real phone number, its purely coincidental.

File: app/src/screens/Dialer/__tests__/Dialer.test.tsx (hunk 4 above)
Comment (kpchoy) [reply app/src/screens/Dialer/__tests__/Dialer.test.tsx:6]: got it!

**Review by kpchoy — APPROVED** [review]

**Pull request merged**